### DISCUSSION TITLE
extract info from description after import

### DISCUSSION THREAD

**Discussion by nick2k3** [firefly-iii/firefly-iii#6913] [Firefly III] [answered]
Hi everyone,
I have just started using Firefly III and I am importing transaction through my bank's export in CSV.
I am able to import and map, however I noticed that my bank places all Apple pay transaction under a similar description:
`PAYMENT VIA MAESTRO                  27-12 24-12-2022 AT 15.58 [Name of merchant] [address of merchant] WITH APPLE PAY [number of Apple Pay card] VIRTUAL CARD NUMBER FOR CONTACTLESS: [number of debit card] CARDHOLDER: [my name]`

the result is that the firefly is assigning all these transaction to the (no name) expense account because it cannot match the name of the merchant from the description (since the descr. is unique due to the date).

I thought of splitting the description in multiple column via excel tricks, but it adds a considerable overhead and head scratches.
Thus I ask: is there a way to post-process the description after the import in Firefly so I can extract the various pieces:
```
PAYMENT VIA MAESTRO                                      --> tag
27-12 24-12-2022 AT 15.58                                 --> date-time or ignored
[Name of merchant] [address of merchant]      --> expense account
WITH APPLE PAY                                                    --> tag
[number of Apple Pay card] VIRTUAL CARD     --> ignored
NUMBER FOR CONTACTLESS: [number of debit card] --> ignored
CARDHOLDER: [my name]                                    --> ignored
```
Thanks,

**Comment by JC5** [comment, marked as answer]
You'll have to do some pre-processing yourself, that would be easiest. And optional thing to do is to use webhooks, where each transaction could be processed by a script you host online somewhere. But both solutions are technically complex.
